src/common/containers/builds.js

```diff
@@ -6,22 +6,27 @@ import BuildHistory from '../components/build-history';
 import { Message } from '../components/forms';
 import Spinner from '../components/spinner';
 
+import { setGitHubRepository } from '../actions/repository-input';
 import { fetchBuilds, fetchSnap } from '../actions/snap-builds';
 
 import styles from './container.css';
 
 class Builds extends Component {
   fetchInterval = null
 
-  fetchData({ snapLink, repoFullName }) {
+  fetchData({ snapLink, repository }) {
     if (snapLink) {
       this.props.dispatch(fetchBuilds(snapLink));
-    } else {
-      this.props.dispatch(fetchSnap(repoFullName));
+    } else if (repository) {
+      this.props.dispatch(fetchSnap(repository.url));
     }
   }
 
-  componentWillMount() {
+  componentDidMount() {
```

Comment:
As a fix for #96 replacing `componentWillMount` with `componentDidMount` to avoid calling it on server.